base/coreimg.jl

```diff
@@ -44,6 +44,13 @@ if !isdefined(Main, :Base)
     (::Type{T}){T}(arg) = convert(T, arg)::T
 end
 
+# Symbol constructors
+Symbol(s::Symbol) = s
+Symbol(s::ASCIIString) = Symbol(s.data)
+Symbol(s::UTF8String) = Symbol(s.data)
+Symbol(a::Array{UInt8,1}) =
+    ccall(:jl_symbol_n, Ref{Symbol}, (Ptr{UInt8}, Int32), a, length(a))
+
```

Comment:
I think most of the builtin type constructors are defined in `boot.jl`. Putting it here should be good also as long as nothing before this depends on it.